pkg/yamlquery/yamlquery_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2023 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package yamlquery_test
+
+import (
+	"testing"
+
+	"github.com/apigee/registry-experimental/pkg/yamlquery"
+	"github.com/google/go-cmp/cmp"
+	"gopkg.in/yaml.v3"
+)
+
+func TestQueryString(t *testing.T) {
+
+	testYaml := `
+sub:
+  list:
+  - one
+  - two
+  map:
+    foo: bar
+    bar: baz
+key: val
+anchored: &anchor
+  foo: bar
+aliased: *anchor`
+
+	node := new(yaml.Node)
+	if err := yaml.Unmarshal([]byte(testYaml), node); err != nil {
+		t.Fatal(err)
+	}
+
+	sp := func(v string) *string { return &v }
+
+	tests := []struct {
+		path string
+		want *string
+	}{
+		{"sub.list.0", sp("one")},
+		{"sub.list.1", sp("two")},
+		// {"sub.list.3", nil}, // panics
+		{"sub.list.not_int", nil},
+		{"sub.list", sp("- one\n- two\n")},
+		{"sub.map.foo", sp("bar")},
+		{"sub.map.bar", sp("baz")},
+		{"sub.map.missing", nil},
+		{"sub.map", sp("foo: bar\nbar: baz\n")},
+		{"key", sp("val")},
+		{"key.is.this.right", sp("val")},
```

Comment:
that's unexpected, but I think I see why (also the reason for the panic) and agree with keeping this PR focused on testing - we can fix those later.

Reply:
Ok. I was concerned that we'd be enshrining this behavior as expected by having tests that confirm it, but if you don't have an issue with it, I'll merge the PR.

Reply:
It let me push! I guess it's supposed to, but that still felt surprising. Anyway, I pushed small fixes for two issues and corresponding test changes.